pkg/operator/configobservation/network/observe_network.go

```diff
@@ -86,26 +90,21 @@ func ObserveServicesSubnet(genericListers configobserver.Listers, recorder event
 
 	previouslyObservedConfig, errs := extractPreviouslyObservedConfig(existingConfig, servicesSubnetConfigPath, bindAddressConfigPath, bindNetworkConfigPath)
 
-	serviceCIDR, err := network.GetServiceCIDR(listers.NetworkLister, recorder)
+	serviceCIDRs, err := GetServiceCIDRs(listers.NetworkLister, recorder)
 	if err != nil {
 		errs = append(errs, err)
 		return previouslyObservedConfig, errs
 	}
+	servicesSubnet := strings.Join(serviceCIDRs, ",")
 
-	if err := unstructured.SetNestedField(out, serviceCIDR, servicesSubnetConfigPath...); err != nil {
+	if err := unstructured.SetNestedField(out, servicesSubnet, servicesSubnetConfigPath...); err != nil {
 		errs = append(errs, err)
 	}
 	bindAddress := "0.0.0.0:6443"
 	bindNetwork := "tcp4"
-	// TODO: only do this in the single-stack IPv6 case once network.GetServiceCIDR()
-	// supports dual-stack
-	if serviceCIDR != "" {
-		if ip, _, err := net.ParseCIDR(serviceCIDR); err != nil {
-			errs = append(errs, err)
```

Comment:
this was validating that the service CIDR was valid and now we no longer validate here.
 I'm not familiar with this code but I assume that was validated before, right?

Reply:
Yes, it is known to already be valid. The code wasn't calling `net.ParseCIDR()` to validate it, it was calling it so it could then call `ip.To4()`, and the author didn't feel comfortable ignoring the error case, even though it "can't happen". The new code _does_ ignore parsing errors (`utilnet.IsIPv6CIDRString` returns `false` if the string doesn't parse, since a string which is not a CIDR is not an IPv6 CIDR). But again, "can't happen" (and would result in more obvious error messages elsewhere anyway if it did happen).